### PR TITLE
engine_rocks: ignore NoSpace error during flush or compaction

### LINKED ISSUE #11224: engine_rocks: NoSpace error during flush or compaction should be ignored
## Development Task

NoSpace error during flush or compaction is harmless, as the output files aren't installed to the db yet.

RocksDB already handles IOError::NoSpace nicely, see https://github.com/tikv/rocksdb/blob/f236fe443d06337ca2538414c62f70ba17b8e409/db/error_handler.cc#L230. We should recognize this case in TIKV's event listener, and do not panic.

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What problem does this PR solve?

Issue Number: close #11224

Problem Summary: Let RocksDB auto recovers from NoSpace error.

### What is changed and how it works?

What's Changed:

The full message for NoSpace error during compaction is like this: "IO Error: No space left on deviceWhile appending to file xxx: No space left on device".

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

### Release note 

```release-note
Do not panic when RocksDB flush or compaction hits disk capacity
```

### CODE REVIEW

File: components/engine_rocks/src/event_listener.rs [hunk 1]
```diff
@@ -95,6 +95,14 @@ impl rocksdb::EventListener for RocksEventListener {
     fn on_background_error(&self, reason: DBBackgroundErrorReason, result: Result<(), String>) {
         assert!(result.is_err());
         if let Err(err) = result {
+            if matches!(
+                reason,
+                DBBackgroundErrorReason::Flush | DBBackgroundErrorReason::Compaction
+            ) && err.contains("No space left")
```
[the remark this answers is not in the document]
Comment (tabokie) [reply components/engine_rocks/src/event_listener.rs:101]: Yes, it's a `Status` variant, see https://github.com/tikv/rocksdb/blob/f236fe443d06337ca2538414c62f70ba17b8e409/util/status.cc#L40

**Review by BusyJay — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**